dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java

```diff
@@ -0,0 +1,444 @@
+
+            String requestorName = requestor.map(authenticatedUser -> authenticatedUser.getFirstName() + " " + authenticatedUser.getLastName())
+                    .orElseGet(() -> BundleUtil.getStringFromBundle("notification.email.info.unavailable"));
+
+            String requestorEmail = requestor.map(AuthenticatedUser::getEmail)
+                    .orElseGet(() -> BundleUtil.getStringFromBundle("notification.email.info.unavailable"));
+
+            messageText += MessageFormat.format(pattern,
+                                                dataFile.getOwner().getDisplayName(), requestorName,
+                                                requestorEmail, getDatasetManageFileAccessLink(dataFile));
+            return messageText;
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String fileMetadataMessage(EmailNotificationDto notificationDto, FileMetadata fileMetadata) {
+        String messageText = BundleUtil.getStringFromBundle("notification.email.greeting");
+
+        if (notificationDto.getNotificationType() == NotificationType.MAPLAYERDELETEFAILED) {
+
+            DatasetVersion version = fileMetadata.getDatasetVersion();
+            String pattern = BundleUtil.getStringFromBundle("notification.email.maplayer.deletefailed.text");
+
+            messageText += MessageFormat.format(pattern,
+                                                fileMetadata.getLabel(), getDatasetLink(version.getDataset()));
+            return messageText;
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String authenticatedUserMessage(EmailNotificationDto notificationDto, String rootDataverseName, InternetAddress systemAddress) {
+        String messageText = BundleUtil.getStringFromBundle("notification.email.greeting");
+
+        if (notificationDto.getNotificationType() == NotificationType.CREATEACC) {
+
+            String accountCreatedMessage = BundleUtil.getStringFromBundle("notification.email.welcome", Arrays.asList(
+                    rootDataverseName,
+                    systemConfig.getGuidesBaseUrl(),
+                    systemConfig.getGuidesVersion(),
+                    BrandingUtil.getSupportTeamName(systemAddress, rootDataverseName),
+                    BrandingUtil.getSupportTeamEmailAddress(systemAddress)
+            ));
+            String optionalConfirmEmailAddon = confirmEmailService.optionalConfirmEmailAddonMsg(notificationDto.getUser());
+            accountCreatedMessage += optionalConfirmEmailAddon;
+            logger.fine("accountCreatedMessage: " + accountCreatedMessage);
+            return messageText + accountCreatedMessage;
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private String getSubjectText(NotificationType notificationType, String rootDataverseName) {
+        List<String> rootDvNameAsList = Collections.singletonList(rootDataverseName);
+        switch (notificationType) {
```

Comment:
Maybe we should make it a method in NotificationType. Subject text is strictly connected with Enum class and frankly this whole switch looks like it doesn't belong here.

Reply:
There are few problems with that:

- The notification type is not tightly correlated with email messages, you can send notification without sending email.

- If I would move extracting subject I would also have to move getMessage method since Subject/Message is directly coupled with each other.

- Both of those methods cannot be extracted to enum since message is using other beans like database access etc.